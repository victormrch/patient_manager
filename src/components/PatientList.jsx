import Patient from "./Patient";

const PatientList = (props) => {
  const { patients, setPatient, deletePatient } = props;

  return (
    <div className="md:w-1/2 lg:w-3/5 md:h-screen overflow-y-scroll">
      {patients && patients.length ? (
        <>
          <h2 className="font-black text-3xl text-center">Patients List</h2>
          <p className="text-xl mt-5 mb-10 text-center">
            Admin your {""}
            <span className="text-indigo-600 font-bold">
              Patients & Appointments
            </span>
          </p>
          {patients.map((patient) => (
            <Patient
              key={patient.id}
              patient={patient}
              setPatient={setPatient}
              deletePatient={deletePatient}
            />
          ))}
        </>
      ) : (
        <>
          <h2 className="font-black text-3xl text-center">No Patients</h2>
          <p className="text-xl mt-5 mb-10 text-center">
            Start adding patients and {""}
            <span className="text-indigo-600 font-bold">
              they will be displayed here
            </span>
          </p>
        </>
      )}
    </div>
  );
};

export default PatientList;
