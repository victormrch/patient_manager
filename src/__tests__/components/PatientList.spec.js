import React from "react";
import { render, screen } from "@testing-library/react";
import PatientList from "../../components/PatientList";
import "@testing-library/jest-dom/extend-expect";

describe("PatientList component specs", () => {
  it("should render correctly using snapshot testing with PatientsList empty", () => {
    const { asFragment } = render(<PatientList />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render correctly using snapshot testing with PatientsList filled", () => {
    const mockPatientList = [
      {
        dischargeDate: "2010-01-01",
        email: "testA@test.com",
        id: "1234",
        name: "Test Name A",
        owner: "Test Owner A",
        reason: "Test Reason A",
      },
      {
        dischargeDate: "2011-01-01",
        email: "testb@test.com",
        id: "12345",
        name: "Test Name B",
        owner: "Test Owner B",
        reason: "Test Reason B",
      },
    ];

    const props = {
      patients: mockPatientList,
      setPatient: jest.fn(),
      deletePatient: jest.fn(),
    };

    const { asFragment } = render(<PatientList {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render No Patients title when PatientsList is empty", () => {
    render(<PatientList />);
    const elementTitle = screen.getByRole("heading", { name: "No Patients" });
    const elementParagraph = screen.getByText("Start adding patients and");

    expect(elementTitle).not.toBeNull();
    expect(elementTitle).toBeInTheDocument();
    expect(elementTitle.tagName).toEqual("H2");
    expect(elementTitle.textContent).toEqual("No Patients");

    expect(elementParagraph).not.toBeNull();
    expect(elementParagraph).toBeInTheDocument();
    expect(elementParagraph.tagName).toEqual("P");
    expect(elementParagraph.textContent).toEqual(
      "Start adding patients and they will be displayed here"
    );
  });
  it("should render No Patients title when PatientsList is empty", () => {
    const mockPatientList = [
      {
        dischargeDate: "2010-01-01",
        email: "testA@test.com",
        id: "1234",
        name: "Test Name A",
        owner: "Test Owner A",
        reason: "Test Reason A",
      },
      {
        dischargeDate: "2011-01-01",
        email: "testb@test.com",
        id: "12345",
        name: "Test Name B",
        owner: "Test Owner B",
        reason: "Test Reason B",
      },
    ];

    const props = {
      patients: mockPatientList,
      setPatient: jest.fn(),
      deletePatient: jest.fn(),
    };

    render(<PatientList {...props} />);
    const elementTitle = screen.getByRole("heading", { name: "Patients List" });
    const elementParagraph = screen.getByText("Admin your");

    expect(elementTitle).not.toBeNull();
    expect(elementTitle).toBeInTheDocument();
    expect(elementTitle.tagName).toEqual("H2");
    expect(elementTitle.textContent).toEqual("Patients List");

    expect(elementParagraph).not.toBeNull();
    expect(elementParagraph).toBeInTheDocument();
    expect(elementParagraph.tagName).toEqual("P");
    expect(elementParagraph.textContent).toEqual(
      "Admin your Patients & Appointments"
    );
  });
});
