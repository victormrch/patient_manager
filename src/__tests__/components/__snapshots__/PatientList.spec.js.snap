// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientList component specs should render correctly using snapshot testing with PatientsList empty 1`] = `
<DocumentFragment>
  <div
    class="md:w-1/2 lg:w-3/5 md:h-screen overflow-y-scroll"
  >
    <h2
      class="font-black text-3xl text-center"
    >
      No Patients
    </h2>
    <p
      class="text-xl mt-5 mb-10 text-center"
    >
      Start adding patients and 
      <span
        class="text-indigo-600 font-bold"
      >
        they will be displayed here
      </span>
    </p>
  </div>
</DocumentFragment>
`;

exports[`PatientList component specs should render correctly using snapshot testing with PatientsList filled 1`] = `
<DocumentFragment>
  <div
    class="md:w-1/2 lg:w-3/5 md:h-screen overflow-y-scroll"
  >
    <h2
      class="font-black text-3xl text-center"
    >
      Patients List
    </h2>
    <p
      class="text-xl mt-5 mb-10 text-center"
    >
      Admin your 
      <span
        class="text-indigo-600 font-bold"
      >
        Patients & Appointments
      </span>
    </p>
    <div
      class="my-5 mx-5 bg-white shadow-md px-5 py-10 rounded-xl"
    >
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Nombre : 
        <span
          class="font-normal normal-case"
        >
          Test Name A
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Owner Name : 
        <span
          class="font-normal normal-case"
        >
          Test Owner A
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Email Address : 
        <span
          class="font-normal normal-case"
        >
          testA@test.com
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        DISCHARGE DATE : 
        <span
          class="font-normal normal-case"
        >
          2010-01-01
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Reason for visit : 
        <span
          class="font-normal normal-case"
        >
          Test Reason A
        </span>
      </p>
      <div
        class="flex flex-row justify-between mt-8"
      >
        <button
          class="py-2 px-10 mb-2 bg-amber-500 hover:bg-amber-700 text-white font-bold uppercase rounded-lg"
          type="button"
        >
          Edit
        </button>
        <button
          class="py-2 px-10 mb-2 bg-red-500 hover:bg-red-700 text-white font-bold uppercase rounded-lg"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
    <div
      class="my-5 mx-5 bg-white shadow-md px-5 py-10 rounded-xl"
    >
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Nombre : 
        <span
          class="font-normal normal-case"
        >
          Test Name B
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Owner Name : 
        <span
          class="font-normal normal-case"
        >
          Test Owner B
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Email Address : 
        <span
          class="font-normal normal-case"
        >
          testb@test.com
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        DISCHARGE DATE : 
        <span
          class="font-normal normal-case"
        >
          2011-01-01
        </span>
      </p>
      <p
        class="font-bold mb-3 text-gray-700 uppercase"
      >
        Reason for visit : 
        <span
          class="font-normal normal-case"
        >
          Test Reason B
        </span>
      </p>
      <div
        class="flex flex-row justify-between mt-8"
      >
        <button
          class="py-2 px-10 mb-2 bg-amber-500 hover:bg-amber-700 text-white font-bold uppercase rounded-lg"
          type="button"
        >
          Edit
        </button>
        <button
          class="py-2 px-10 mb-2 bg-red-500 hover:bg-red-700 text-white font-bold uppercase rounded-lg"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
